pkg/member/member_control.go

```diff
@@ -317,10 +318,17 @@ func (m *memberControl) IsLearnerPresent(ctx context.Context) (bool, error) {
 
 // IsClusterScaledUp determines whether a etcd cluster is getting scale-up or not and returns a boolean
 func (m *memberControl) IsClusterScaledUp(ctx context.Context, clientSet client.Client) (bool, error) {
-	state, err := miscellaneous.GetInitialClusterStateIfScaleup(ctx, m.logger, clientSet, m.podName, m.podNamespace)
+	m.logger.Info("Checking whether etcd cluster is marked for scale-up")
+	curSts := &appsv1.StatefulSet{}
+	err := clientSet.Get(ctx, client.ObjectKey{
+		Namespace: m.podNamespace,
+		Name:      m.podName,
+	}, curSts)
```

Comment:
This is getting repeated in 2 places.
Have a method:
```go
// GetStatefulSet gets the StatefulSet with the name podName in podNs namespace. It will return if there is any error or the StatefulSet is not found.
func GetStatefulSet(ctx context.Context, cli client.Client, podNamespace string, podName string) (*appsv1.StatefulSet, error) {
	curSts := &appsv1.StatefulSet{}
	if err := cli.Get(ctx, client.ObjectKey{Namespace: podNamespace, Name: podName[:strings.LastIndex(podName, "-")]}, curSts); err != nil {
		return nil, err
	}
	return curSts, nil
}
```

Reply:
In `GetInitialClusterStateIfScaleup` you can then call:
```go
	sts, err := GetStatefulSet(ctx, clientSet, podNamespace, podName)
	if err != nil {
		logger.Errorf("unable to fetch sts {namespace: %s, name: %s} %v", podNamespace, podName, err)
		return nil, err
	}

	if metav1.HasAnnotation(sts.ObjectMeta, ScaledToMultiNodeAnnotationKey) {
		return pointer.StringPtr(ClusterStateExisting), nil
	}
```

Reply:
And in member_control.go/IsClusterScaleup you can have:
```go
	sts, err := miscellaneous.GetStatefulSet(ctx, clientSet, m.podNamespace, m.podName)
	if err != nil {
		m.logger.Errorf("unable to fetch sts {namespace: %s, name: %s} %v", m.podNamespace, m.podName, err)
	} else {
		if metav1.HasAnnotation(sts.ObjectMeta, miscellaneous.ScaledToMultiNodeAnnotationKey) {
			return true, nil
		}
	}
	isEtcdMemberPresent, err := m.IsMemberInCluster(ctx)
	if err != nil || isEtcdMemberPresent {
		return false, err
	}
	return true, nil

```